### PR TITLE
Adapt GetDestinations to be unit tested

### PR DESCRIPTION
**Pull request by skyylex**
### Description

General refactoring and decomposition of the `GetDestinations` to prepare for unit tests

### Details

The initial idea was to just separate part related to CLI to complete unit tests for GetDestinations. However, there were some minor things that was possible to improve like making the logic more strict (using more enums!) or decomposing (parse, command protocol and struct). Subjectively, it's a bit more structured at the moment. Objectively it's possible to write unit tests with mocked environment.

### CODE REVIEW

File: Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift [hunk 1]
```diff
@@ -8,11 +8,17 @@
...
-public class GetDestinations {
-    public init() {}
+/// An abstraction over a bash script that executes and prints an output
+public protocol ConsoleCommand {
+    func execute(with: Workflow) -> String?
+}
 
-    public func getAvailable(workflow: Workflow) throws -> [Destination] {
-        let string = try CommandLine().runBash(
+/// Runs xcrun ang get list of available iOS simulators
```
Comment (skyylex) [inline Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift]: Runs `xcrun` ang get list of available simulators

Comment (onmyway133) [reply Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift]: Nice spot

File: Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift [hunk 2]
```diff
@@ -8,11 +8,17 @@
 import PumaCore
 import Foundation
 
-public class GetDestinations {
-    public init() {}
+/// An abstraction over a bash script that executes and prints an output
+public protocol ConsoleCommand {
```
Comment (onmyway133) [inline Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift]: I'm not particularly fond of using protocol ...

Comment (skyylex) [reply Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift]: I don't have a preference here. I could change to a function injection. Just need a way to mock bash call. Something like:

```
public struct GetListOfDevicesCommand {
    public static func execute(with workflow: Workflow) -> String? {
       //
    }
}

public class GetDestinations {
    public typealias BashCommand = (Workflow) -> String?
    private let bashCommand: BashCommand

    public init(bashCommand: BashCommand = GetListOfDevicesCommand.execute) {
    
    }
}
```

Comment (onmyway133) [reply Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift]: Ja I see what you want to achieve, maybe using function injection is better, as introducing protocol gives more overhead

File: Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift [hunk 3]
```diff
@@ -24,81 +24,106 @@ public class GetDestinations {
...
+    }
+    
+}
 
-        guard let data = string.data(using: .utf8) else {
+public class GetDestinations {
+    public typealias BashCommand = (Workflow) -> String?
+    private let bashCommand: BashCommand
+    
+    public init(bashCommand: @escaping BashCommand = GetListOfDevicesCommand.execute) {
```
Comment (onmyway133) [inline Sources/iOS/Library/Get/GetDestinations/GetDestinations.swift]: Looks good to me 👍

**Pull request merged**